frontend/src/scenes/retention/retentionTableLogic.tsx

```diff
@@ -86,15 +87,41 @@ export const retentionTableLogic = kea<retentionTableLogicType>({
             ({ filters }): Partial<FilterType> => (filters?.insight === InsightType.RETENTION ? filters ?? {} : {}),
         ],
         results: [
+            // Take the insight result, and cast it to `RetentionTablePayload[]`
             (s) => [s.insight],
-            ({ filters, result }): RetentionTablePayload[] | RetentionTrendPayload[] => {
-                return filters?.insight === InsightType.RETENTION &&
-                    result &&
-                    (result.length === 0 ||
-                        (!result[0].values && filters.display === ACTIONS_LINE_GRAPH_LINEAR) ||
-                        (result[0].values && filters.display === ACTIONS_TABLE))
-                    ? result
-                    : []
+            ({ filters, result }): RetentionTablePayload[] => {
+                return filters?.insight === InsightType.RETENTION && result
+            },
+        ],
+        trendSeries: [
```

Comment:
So this is quite non-obvious code IMO that's lacking any tests. 

I need a bit of context - why are we doing this in the frontend rather than the backend? This would keep the FE free of this business logic and we'd have good tools for testing this together with the real data.